src/swap/index.ts

```diff
@@ -981,6 +984,108 @@ class Swap extends Tx {
+      const poolData = await this.getPoolData(poolAddress)
+      const { taxman: taxmanAddress } = poolData
+      if (!account.isAddress(taxmanAddress)) throw ProgramError.InvalidTaxmanAddressErr
+      const taxmanPublicKey = account.fromAddress(taxmanAddress) as PublicKey
+
+      const srcAddress = ra.srcAddress
+      if (!account.isAddress(srcAddress)) throw ProgramError.InvalidAddressErr
+      const srcPublicKey = account.fromAddress(srcAddress) as PublicKey
+      const { mint: srcMintAddress } = await this._splt.getAccountData(srcAddress)
+      if (!account.isAddress(srcMintAddress)) throw ProgramError.InvalidSourceMintAddressErr
+      const srcMintPublicKey = account.fromAddress(srcMintAddress) as PublicKey
+
+      const dstAddress = ra.dstAddress
+      if (!account.isAddress(dstAddress)) throw  ProgramError.InvalidAddressErr
+      const dstPublicKey = account.fromAddress(dstAddress) as PublicKey
+      const { mint: dstMintAddress } = await this._splt.getAccountData(dstAddress)
+      if (!account.isAddress(dstMintAddress)) throw ProgramError.InvalidDestinationMintAddressErr
+      const dstMintPublicKey = account.fromAddress(dstMintAddress) as PublicKey
+
+      const treasuryTaxmanAddress = await this._splt.deriveAssociatedAddress(taxmanAddress, dstMintAddress)
+      const treasuryTaxmanPublicKey = account.fromAddress(treasuryTaxmanAddress) as PublicKey
+
+      const [treasuryBidPublicKey, treasuryAskPublicKey] = [
+        this.findTreasury(srcMintAddress, poolData),
+        this.findTreasury(dstMintAddress, poolData),
+      ].map(
+        (treasuryAddress) => account.fromAddress(treasuryAddress) as PublicKey,
+      )
+
+      keys.push({ pubkey: poolPublicKey, isSigner: false, isWritable: true })
+
+      keys.push({ pubkey: srcPublicKey, isSigner: false, isWritable: true })
+      keys.push({ pubkey: srcMintPublicKey, isSigner: false, isWritable: false })
+      keys.push({ pubkey: treasuryBidPublicKey, isSigner: false, isWritable: true })
+
+      keys.push({ pubkey: dstPublicKey, isSigner: false, isWritable: true })
+      keys.push({ pubkey: dstMintPublicKey, isSigner: false, isWritable: false })
+      keys.push({ pubkey: treasuryAskPublicKey, isSigner: false, isWritable: true })
+
+      keys.push({ pubkey: taxmanPublicKey, isSigner: false, isWritable: false })
+      keys.push({ pubkey: treasuryTaxmanPublicKey, isSigner: false, isWritable: true })
+
+      keys.push({ pubkey: treasurerPublicKey, isSigner: false, isWritable: false })
+    }
+
+    let transaction = new Transaction()
+    transaction = await this.addRecentCommitment(transaction)
+    const layout = new soproxABI.struct(
+      [
+        { key: 'code', type: 'u8' },
+        { key: 'amount', type: 'u64' },
+        { key: 'limit', type: 'u64' },
+      ],
+      {
+        code: InstructionCode.Routing.valueOf(),
+        amount,
+        limit,
+      },
+    )
+    const instruction = new TransactionInstruction({
```

Comment:
Testing the limit computation before sending the real action. Refer `initializePool` for example.